Zend/zend_vm_def.h

```diff
@@ -8238,4 +8238,397 @@ ZEND_VM_HANDLER(183, ZEND_BIND_STATIC, CV, CONST, REF)
 	ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
 }
 
+ZEND_VM_TYPE_SPEC_HANDLER(ZEND_ADD, ((res_info & (MAY_BE_ANY|MAY_BE_UNDEF)) == MAY_BE_LONG && (op1_info & (MAY_BE_ANY|MAY_BE_UNDEF)) == MAY_BE_LONG && (op2_info & (MAY_BE_ANY|MAY_BE_UNDEF)) == MAY_BE_LONG), ZEND_ADD_LONG_NO_OVERFLOW, CONST|TMPVARCV, CONST|TMPVARCV, SPEC(NO_CONST_CONST,COMMUTATIVE))
```

Comment:
the last version use simpler conditions


________________________________
From: Andrea Faulds <notifications@github.com>
Sent: Thursday, March 17, 2016 21:00
To: php/php-src
Cc: Dmitry Stogov
Subject: Re: [php-src] Transparently introduce type-specialized opcode handlers. (#1824)


In Zend/zend_vm_def.h<https://github.com/php/php-src/pull/1824#discussion_r56549720>:

> @@ -8238,4 +8238,397 @@ ZEND_VM_HANDLER(183, ZEND_BIND_STATIC, CV, CONST, REF)
>       ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
>  }
>
> +ZEND_VM_TYPE_SPEC_HANDLER(ZEND_ADD, ((res_info & (MAY_BE_ANY|MAY_BE_UNDEF)) == MAY_BE_LONG && (op1_info & (MAY_BE_ANY|MAY_BE_UNDEF)) == MAY_BE_LONG && (op2_info & (MAY_BE_ANY|MAY_BE_UNDEF)) == MAY_BE_LONG), ZEND_ADD_LONG_NO_OVERFLOW, CONST|TMPVARCV, CONST|TMPVARCV, SPEC(NO_CONST_CONST,COMMUTATIVE))


I would agree it's too verbose. It's too long to read quickly and this means it's easier to miss errors.

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/php/php-src/pull/1824/files/9020086a13c506ec127a648d08e9a30a81f08883#r56549720>